src/tests/utils.cairo

```diff
@@ -436,85 +494,143 @@ fn accelerate_to_running(ref self: VaultFacade) {
+self.timeskip_and_settle_round();
+}
 
-    let index: u32 = 0;
-    while index < bidders {
-        let bid_quant = (bid_amount / bidders.into()) + 1;
-        let bid_price = params.reserve_price + index.into();
-        current_round.place_bid(bid_amount / 2, bid_price, *bidders_array.at(index));
-    }
+
+//Create various amounts array (For bids use the function twice for price and amount)
+fn create_array_linear(amount:u256, len:u32)->Array<u256>{
+
+    let mut arr:Array<u256> = array![];
+    let mut index = 0;
+    while (index<len) {
+        arr.append(amount);
+        index+=1;
+    };
+    arr
 }
 
+fn create_array_gradient(amount:u256,step:u256, len:u32) ->Array<u256>{
+
+    let mut arr:Array<u256> = array![];
+    let mut index:u32 = 0;
+    while(index<len){
+        arr.append(amount+index.into()*step);
+        index+=1;
+    };
+    arr
+}
+
+
+// fn accelerate_to_auctioning_n_linear(ref self: VaultFacade, providers: u32, amount:u256){
+//     deposit_n(ref self,providers,amount);
+//     self.start_auction();
+// }
+
+
+// fn accelerate_to_auctioning_n_custom(ref self: VaultFacade, providers: u32, amounts: Array<u256>) {
+
+//     deposit_n_custom(ref self, providers, amounts);
+//     self.start_auction();
+// }
+
+
+// //Auction with linear bidding
+// fn accelerate_to_running_n_linear(ref self: VaultFacade, providers: u32, bidders: u32,amount:u256) -> u256 {
+//     let mut current_round = self.get_current_round();
+//     if (current_round.get_state() != OptionRoundState::Auctioning) {
+//         accelerate_to_auctioning(ref self);
+//     }
+//     let mut current_round = self.get_current_round();
+//     let params = current_round.get_params();
+//     let bid_amount = params.total_options_available;
+//     let bid_price = params.reserve_price;
+//     let bid_amount = bid_amount * bid_price / bidders.into();
+//     bid_n(ref self,bidders,bid_amount,bid_price);
+//     set_block_timestamp(params.auction_end_time + 1);
+//     current_round.end_auction();
+//     bid_amount
+// }
+
+
+// //Applies a gradient to bid price;
+
+
+// fn accelerate_to_running_n_custom(
+//     ref self: VaultFacade, providers: u32, bidders: u32, amounts: Array<u256>, prices: Array<u256>
+// ) {
+//     let mut current_round = self.get_current_round();
+//     if (current_round.get_state() != OptionRoundState::Auctioning) {
+//         accelerate_to_auctioning(ref self);
+//     }
+//     let params = current_round.get_params();
+//     bid_n_custom(ref self,bidders,amounts,prices);
+//     set_block_timestamp(params.auction_end_time + 1);
+//     current_round.end_auction();
+// }
+
 //Auction with partial bidding
-fn accelerate_to_running_n_partial(ref self: VaultFacade, providers: u32, bidders: u32) {
-    accelerate_to_auctioning_n(ref self, providers);
-    let bidders_array: Array<ContractAddress> = option_bidders_get(bidders);
+
+
+
+//@dev Should we create more complex helpers for creating conditions like this directly?
```

Comment:
i think we can just use the single accelerator, then modify the arrays we pass to it to replicate this